var/spack/repos/builtin/packages/py-tensorflow/package.py

```diff
@@ -147,12 +149,18 @@ class PyTensorflow(Package, CudaPackage, ROCmPackage, PythonExtension):
     )
 
     extends("python")
-    depends_on("python@3:", type=("build", "run"), when="@2.1:")
-    # https://github.com/tensorflow/tensorflow/issues/33374
-    depends_on("python@:3.7", type=("build", "run"), when="@:2.1")
+
+    # Python support based on wheel availability
```

Comment:
I don't normally like doing this, but in my experience, libraries like this are so tightly coupled to CPython internals that they don't build for newer versions without manual effort. This also gives us an easy way to deprecate ancient versions of TF that no one uses anymore. Did the same for ancient versions of bazel that TF used to require. Hopefully this makes these packages more manageable.